File: Leetcode solutions/0007-reverse-integer/0007-reverse-integer.py
class Solution:
    def reverse(self, x: int) -> int:
        sum=0
        sign=1
        if x<0:
            sign=-1
            x=x*-1
        while x>0:
            rem=x % 10
            sum=(sum*10)+rem
            x=x//10
        if not -2147483648<sum<2147483647:
            return 0
        return sign*sum

File: Leetcode solutions/0007-reverse-integer/0007-reverse-integer.java
class Solution {
    public int reverse(int x) {
        int rem;
        long rev = 0;
        while(x!=0){
            rem = x%10;
            rev = rev*10+rem;
            x=x/10;
        }
        return (rev<Integer.MIN_VALUE|rev>Integer.MAX_VALUE)?0:(int)rev;
    }
}
